src/python/pants/core/goals/test.py

```diff
@@ -181,6 +181,7 @@ class TestDebugRequest:
 
 
 @union
+@dataclass(frozen=True)
```

Comment:
This was necessary to expose `sources` in the ctor. Previously this was papered over by the fact that we didn't try to pass `sources` to the ctor due to them being silently ignored because of the type hint issue.

Reply:
Oopsies. Thanks!